src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -188,8 +188,11 @@ internal partial class AnalyzerManager
 
                 var handler = new EventHandler<Exception>((sender, ex) =>
                     {
-                        var diagnostic = AnalyzerExecutor.GetAnalyzerDiagnostic(analyzer, ex);
-                        analyzerExecutor.OnAnalyzerException?.Invoke(ex, analyzer, diagnostic);
+                        var diagnostic = AnalyzerExecutor.GetAnalyzerExceptionDiagnostic(analyzer, ex);
+                        if (diagnostic != null)
```

Comment:
This pattern appears in several places. I have no idea how many different methods are passed as onAnalyzerException. If there are not many, and if we control all of them, it might be more robust to put the null check inside these methods.

I sign off.